Update to module template

adonesky1 opened the pull request:
The GitHub actions have been updated to match the module template. 

There are still differences because this library is not yet using TypeScript, so it doesn't have any build step. 

It also doesn't yet have API documentation, so steps relating to that have been removed.

adonesky1 commented on .github/workflows/publish-release.yml:
```diff
@@ -1,29 +1,79 @@
+        with:
+          # This `NPM_TOKEN` needs to be manually set per-repository.
+          # Look in the repository settings under "Environments", and set this token in the `npm-publish` environment.
+          npm-token: ${{ secrets.NPM_TOKEN }}
```
I believe @kumavis can add this token... @Gudahtt not sure if you can as well?

Gudahtt replied:
Yep, I can set this up

adonesky1 commented:
@Gudahtt can we remove the `Require Additional Reviewer for Releases` workflow now?

mcmire commented on .github/workflows/lint-test.yml:
```diff
@@ -1,47 +1,74 @@
-name: Lint, and Test
+name: Lint and Test
```
This file got renamed recently to `build-lint-test.yml`.

Also, the `name` should be "Build, Lint, and Test".

adonesky1 replied:
Ok! Yeah I modified because there isn't actually a build step for the repo yet, but I see why we'd want to keep it standardized

mcmire reviewed (commented):
The changes here look good. One new thing about the workflow files is that there is now a [`main.yml` file](https://github.com/MetaMask/metamask-module-template/blob/main/.github/workflows/main.yml) which kicks everything off — the other files represent workflows that are spawned from this workflow. You should be able to copy this file straight from the module template.

mcmire commented:
@adonesky1 Also, with regard to the "Require Additional Reviewer" workflow, [we did remove that from the module template recently](https://github.com/MetaMask/metamask-module-template/pull/148) so I think it's safe to remove here.

adonesky1 commented:
Addressed your feedback [here](https://github.com/MetaMask/eth-simple-keyring/pull/123/commits/47ddcc9af3aaf3b791cfce16d76c5160782c8739). Thanks @mcmire!

Gudahtt commented on .github/workflows/publish-release.yml:
```diff
@@ -1,29 +1,79 @@
+      - name: Install
+        run: |
+          yarn install
+          yarn build
```
Whoops - left a build step in here

Gudahtt replied:
A bunch of this workflow is centered around building, then caching that build, so it wouldn't apply here.

Check out this repository for a more recent example of how to set this up in a repo with no build step: https://github.com/MetaMask/KeyringController/blob/main/.github/workflows/publish-release.yml

adonesky1 replied:
Ok thanks! Should the root level `on` be:
```
on:
  push:
    branches: [main]
```
as it is in the KeyringController example: https://github.com/MetaMask/KeyringController/blob/main/.github/workflows/publish-release.yml#L3
or
```
on:
  workflow_call:
    secrets:
      NPM_TOKEN:
        required: true
```    
as it is in the `module-template` repo: https://github.com/MetaMask/metamask-module-template/blob/main/.github/workflows/publish-release.yml#L3

adonesky1 replied:
Oh I see so the one on the `keyringController` runs for each push to `main` but first checks if its a release branch when deciding whether to continue?

Gudahtt replied:
The module template way of structuring the workflows is perfectly fine here. Sorry for the confusion, that repo was last updated before that restructuring was done.
